### ISSUE TITLE
[FEA] Binning/Histograms

### ISSUE THREAD

**Issue by quasiben** [rapidsai/cudf#1061]
In exploration of data, histogramming with a fixed bin size can be useful when trying to understand distribution and especially skew.  I would like to be able to generate a histogram on a column/series of a dataframe.  While `value_counts` exists, it doesn't expose an option to set the number of bins.  Providing both an option to value_counts and a new method, `hist` would be valuable.

cc @thomcom

**Comment by jrhemstad** [comment]
We could probably provide this at the libcudf level.

**Comment by beckernick** [comment]
We could create stopgap functionality for this feature at the Python level if current users need this. The following example code works as long as the minimum value in the bin array is less than the minimum value in the data array.

```python
def histogram(ser, bins):
    binned = ser.digitize(bins, right=False)
    vc = binned.value_counts().sort_index()
    return vc, cudf.Series(bins)

nelem = 100000
bins = np.array([-20, 0, 3, 6, 9, 10, 19., 50])
arr = np.random.normal(5,5,nelem)
s = cudf.Series(arr)
hist_np = np.histogram(arr, bins)
hist_cudf = histogram(s, bins)
np.testing.assert_array_almost_equal(hist_np[0], hist_cudf[0])
```

**Comment by jakirkham** [comment]
Should be able to keep things on the device with CuPy.

https://docs-cupy.chainer.org/en/stable/reference/generated/cupy.histogram.html

**Comment by rgsl888prabhu** [comment]
Planning to use cupy support this feature.

**Comment by jakirkham** [comment]
Cool, thanks Ram! Please let us know if you run into any issues or need another pair of eyes 🙂

**Comment by kkraus14** [comment]
We recently implemented some binning functionality in libcudf and is being worked on now to plumb into `cudf.cut` here: https://github.com/rapidsai/cudf/pull/8002

**Comment by beckernick** [comment]
I think this should be considered closed by the combination of #8002 and #8247 (when it lands). `pd.DataFrame.hist` creates a plot of the histogram, which I think is separate from this issue and can be filed independently if desired

**Comment by vyasr** [comment]
I agree, I think multidimensional histogramming or any more general functionality than #8002 is out of scope for `cudf` and I don't know that we plan to offer anything along the lines of the matplotlib plotting functionality given by `pandas.DataFrame.hist`.
